vividus-plugin-mobitru/src/main/java/org/vividus/mobitru/MobitruFacade.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2019-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.vividus.mobitru;
+
+import java.io.IOException;
+import java.time.Duration;
+import java.util.List;
+import java.util.Objects;
+
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import org.apache.commons.lang3.Validate;
+import org.openqa.selenium.remote.DesiredCapabilities;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.vividus.http.client.ExternalServiceException;
+import org.vividus.mobitru.model.Application;
+import org.vividus.mobitru.model.Device;
+import org.vividus.util.wait.DurationBasedWaiter;
+import org.vividus.util.wait.WaitMode;
+import org.vividus.util.wait.Waiter;
+
+public class MobitruFacade
+{
+    private static final Logger LOGGER = LoggerFactory.getLogger(MobitruFacade.class);
+
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper()
+            .disable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES);
+
+    private static final int RETRY_TIMES = 20;
+
+    private final MobitruClient mobitruClient;
+    private Duration waitForDeviceTimeout;
+
+    public MobitruFacade(MobitruClient mobitruClient)
+    {
+        this.mobitruClient = mobitruClient;
+    }
+
+    public String takeDevice(DesiredCapabilities desiredCapabilities) throws IOException
```

Comment:
```suggestion
    public String takeDevice(DesiredCapabilities desiredCapabilities) throws MobitruOperationException
```
it's better to introduce a new exception type for the facade